lib/src/rules/avoid_shadowing_type_parameters.dart

```diff
@@ -59,6 +60,11 @@ class _Visitor extends SimpleAstVisitor<void> {
     _checkAncestorParameters(functionExpression.typeParameters, node);
   }
 
+  @override
+  void visitGenericTypeAlias(GenericTypeAlias node) {
+    _checkForShadowing(node.functionType.typeParameters, node.typeParameters);
```

Comment:
`_checkForShadowing` doesn't check the first parameter to ensure that it isn't `null`. We should either change that (and clean up existing code) or add a null check here.

Reply:
Ah, good catch. I think I like null-checking before the function here. Done.